pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/creator/RangeIndexCreatorTest.java

```diff
@@ -106,23 +109,22 @@ private void testDataType(DataType dataType)
       throws IOException {
     FieldSpec fieldSpec = new DimensionFieldSpec(COLUMN_NAME, dataType, true);
     int numDocs = 1000;
-    Number[] values = new Number[numDocs];
+    Object values = valuesArray(dataType, numDocs);
 
+    int numValuesPerRange;
     try (RangeIndexCreator creator = new RangeIndexCreator(INDEX_DIR, fieldSpec, dataType, -1, -1, numDocs, numDocs)) {
       addDataToIndexer(dataType, numDocs, 1, creator, values);
       creator.seal();
+      // account for off by one bug in v1 implementation
+      numValuesPerRange = creator.getNumValuesPerRange() + 1;
```

Comment:
The current implementation reports 50 values per bucket but assigns 51 values to the first 19 buckets and 31 to the last one.

Reply:
I see, you are right. The `if (i > start + boundary)` in the current implementation is causing this, which should be `if (i >= start + boundary)` instead.
Since we're going to deprecate the old implementation, okay to leave it as is